File: src/languages/python/binary/gates.py
def Not(a) :
	o = True
	if a == True:
		o=False
	return o

def And(a, b) :
	o = False
	if a == True and b == True:
		o=True
	return o

def Nand(a, b) :
	return Not(And(a, b))


def Nor(a, b) :
	return And(Not(a), Not(b))


def Or(a, b) :
	return Not(And(Not(a), Not(b)))


def Xor(a, b) :
	return Or(And(a, Not(b)), And(Not(a), b))


def Xnor(a, b) :
	return Not(Xor(a, b))

File: src/languages/c#/binary/gates.cs
namespace minicallib_gates{
    public class gates {
        
        public static boolean Not(boolean a) {
            boolean o = true;
            if (a == true){
                o=false;
            }
            return o;
        }
        
        public static boolean And(boolean a, boolean b) {
            boolean o = false;
            if (a == true && b == true){
                o = true;
            }
            return o;
        }
        
        public static boolean Nand(boolean a, boolean b) {
            return Not(And(a, b));
        }
        
        public static boolean Nor(boolean a, boolean b) {
            return And(Not(a), Not(b));
        }
        
        public static boolean Or(boolean a, boolean b) {
            return Not(And(Not(a), Not(b)));
        }
        
        public static boolean Xor(boolean a, boolean b) {
            return Or(And(Not(a), b), And(a, Not(b)));
        }
        
        public static boolean Xnor(boolean a, boolean b) {
            return Not(Or(And(Not(a), b), And(a, Not(b))));
        }
        

    }
}